forte/data/data_pack.py

```diff
@@ -1597,28 +1619,58 @@ def require_annotations(entry_class=Annotation) -> bool:
             yield from []
             return
 
+        # If range_annotation is specified, we record its begin and
+        # end index
+        range_begin: int
+        range_end: int
+
+        if range_annotation is not None:
+            if isinstance(range_annotation, (Annotation, AudioAnnotation)):
+                range_begin = range_annotation.begin
+                range_end = range_annotation.end
+            else:
+                # range_annotation is given by the tid of the entry it
+                # represents
+                range_raw = self._data_store.transform_data_store_entry(
+                    self.get_entry_raw(range_annotation)
+                )
+                range_begin = range_raw[BEGIN_ATTR_NAME]
+                range_end = range_raw[END_ATTR_NAME]
+
         try:
             for entry_data in self._data_store.get(
-                type_name=get_full_module_name(entry_type_),
+                type_name=entry_type_,
                 include_sub_type=include_sub_type,
                 range_span=range_annotation  # type: ignore
-                and (range_annotation.begin, range_annotation.end),
+                and (range_begin, range_end),
             ):
-                entry: Entry = self.get_entry(tid=entry_data[TID_INDEX])
+
                 # Filter by components
                 if components is not None:
-                    if not self.is_created_by(entry, components):
+                    if not self.is_created_by(
+                        entry_data[TID_INDEX], components
+                    ):
                         continue
 
-                # Filter out incompatible audio span comparison for Links and Groups
-                if (
-                    issubclass(entry_type_, (Link, Group))
-                    and isinstance(range_annotation, AudioAnnotation)
-                    and not self._index.in_audio_span(
-                        entry, range_annotation.span
+                entry: Union[Entry, Dict[str, Any]]
+                if get_raw:
+                    entry = self._data_store.transform_data_store_entry(
+                        entry_data
                     )
-                ):
-                    continue
+                else:
+                    entry = self.get_entry(tid=entry_data[TID_INDEX])
+
+                    # Filter out incompatible audio span comparison for Links and Groups
+                    if (
```

Comment:
Maybe this filtering also applies to cases when `get_raw==True`?

Reply:
Yes it does. But I felt that this change is currently specific to entries who's objects exist. Changing methods like `in_span` and `in_audio_span` would make this PR a little big. That's why I felt like we could modify `in_span` and `in_audio_span` in another PR.

Reply:
Okay then let's create an issue for that

Reply:
Done: #901